server/src/main/java/org/apache/druid/server/coordinator/helper/DruidCoordinatorRuleRunner.java

```diff
@@ -84,8 +85,10 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
     // find available segments which are not overshadowed by other segments in DB
     // only those would need to be loaded/dropped
     // anything overshadowed by served segments is dropped automatically by DruidCoordinatorCleanupOvershadowed
-    final Set<DataSegment> overshadowed = ImmutableDruidDataSource
-        .determineOvershadowedSegments(params.getAvailableSegments());
+    // If metadata store hasn't been polled yet, use empty overshadowed list
+    final Collection<DataSegment> overshadowed = Optional
```

Comment:
Keep this as a Set, please. Later on, we're doing `overshadowed.contains(segment)`, and that's scary if `overshadowed` is any kind of Collection other than a Set (or very short List).

By the way, it would be clearer to make this a `Set<SegmentId>` rather than `Set<DataSegment>`.

Reply:
changed to Set